adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -338,6 +338,9 @@ private Bundle getAuthTokenFromAccountManager(final AuthenticationRequest reques
                 Logger.e(TAG, AUTHENTICATOR_CANCELS_REQUEST, "", ADALError.AUTH_FAILED_CANCELLED, e);
             } catch (AuthenticatorException e) {
                 Logger.e(TAG, AUTHENTICATOR_CANCELS_REQUEST, "", ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING);
+                if (!StringExtensions.isNullOrBlank(e.getMessage()) && e.getMessage().contains(ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE.toString())) {
```

Comment:
Yes, I will add the network checking at the beginning of talking to broker in the following commit, which will cover both the bound service situation and the old way. 

For Force_prompt.Force_prompt, shouldTrySilentFlow(authenticationRequest) will return false, the auto process will go to the acquireTokenInteractiveFlow() where the network connection is checked at the beginning.